src/blas/backends/rocblas/rocblas_scope_handle.hpp

```diff
@@ -0,0 +1,93 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+#ifndef ROCBLAS_SCOPED_HANDLE_HIPSYCL_HPP
+#define ROCBLAS_SCOPED_HANDLE_HIPSYCL_HPP
+#include <CL/sycl.hpp>
+#include <memory>
+#include <thread>
+#include <atomic>
+#include <unordered_map>
+
+#include "rocblas_helper.hpp"
+#include <CL/sycl/detail/backend_traits_hip.hpp>
+namespace oneapi {
+namespace mkl {
+namespace blas {
+namespace rocblas {
+
+template <typename T>
+struct rocblas_handle_container {
+    using handle_container_t = std::unordered_map<T, std::atomic<rocblas_handle> *>;
+    handle_container_t rocblas_handle_container_mapper_{};
+    ~rocblas_handle_container() noexcept(false);
+};
+
+/**
+* @brief NVIDIA advise for handle creation:
```

Comment:
Are these comments still valid for rocBLAS backend?

Reply:
Yes, this is true ([ref](https://rocblas.readthedocs.io/en/rocm-4.3.0/handle_stream_device.html#stream-and-device-management)).

Reply:
The comments need to be adjusted accordingly for rocBLAS. We'll add relevant info to this.